src/routes/semrush-route.php

```diff
@@ -95,4 +150,24 @@ public function authenticate( WP_REST_Request $request ) {
 	public function has_valid_code( $code ) {
 		return $code !== '';
 	}
+
+	/**
+	 * Checks if a valid country code was submitted.
+	 *
+	 * @param string $country_code The country code to check.
+	 *
+	 * @return boolean Whether or not the country code is valid.
+	 */
+	public function has_valid_country_code( $country_code ) {
+		return ( $country_code !== '' && preg_match( '/^[a-z]{2}$/', $country_code ) === 1 );
+	}
+
+	/**
+	 * Whether or not the current user is allowed to edit post and thus access the SEMrush modal.
+	 *
+	 * @return boolean Whether or not the current user is allowed to edit posts.
+	 */
+	public function can_edit() {
+		return \current_user_can( 'edit_posts' );
```

Comment:
Question: has the capability to check against been agreed with product?

Reply:
I chose the `edit_posts` capability since it's the one that's need to access the Edit Post screen and then the Yoast metabox/sidebar from which you can open the SEMrush modal with the country selector.
Checking for the capability also enforce the authentication check on the user.
@thijsdevalk what do you think? Could this be the right capability to check?

Reply:
Good question! Is this just about about the country selector? If that's the case, then editor is fine. If this is about the entire SEMrush flow, including the authorisation, then it's not :) 

The authorisation flow should be limited to admins. If that, for now at least, means that the _entire_ SEMrush flow/functionality is limited to admins, then that's ok. But we'll probably want to limit just the authorisation flow later on to admins, and once the site is authorised make sure other users, such as those with edit capabilities, can use the tool as well.

Reply:
Ok, leaving that as it is since it's just for setting the country code option via REST API, and probably the tool will be accessible to users who can write posts in the future.